doc/index.rst

```diff
@@ -1,40 +1,93 @@
-=======
-Welcome
-=======
-
-.. epigraph:: Sphinx makes it easy to create intelligent and beautiful documentation.
-
-Here are some of Sphinx's major features:
-
-* **Output formats:** HTML (including Windows HTML Help), LaTeX (for printable
-  PDF versions), ePub, Texinfo, manual pages, plain text
-* **Extensive cross-references:** semantic markup and automatic links for
-  functions, classes, citations, glossary terms and similar pieces of
-  information
-* **Hierarchical structure:** easy definition of a document tree, with automatic
-  links to siblings, parents and children
-* **Automatic indices:** general index as well as a language-specific module
-  indices
-* **Code handling:** automatic highlighting using the Pygments_ highlighter
-* **Extensions:** automatic testing of code snippets, inclusion of docstrings
-  from Python modules (API docs) via :ref:`built-in extensions
-  <builtin-extensions>`, and much more functionality via :ref:`third-party
-  extensions <third-party-extensions>`.
-* **Themes:** modify the look and feel of outputs via :doc:`creating themes
-  <development/theming>`, and reuse many :ref:`third-party themes
-  <third-party-themes>`.
-* **Contributed extensions:** dozens of extensions :ref:`contributed by users
-  <third-party-extensions>`; most of them installable from PyPI.
+======
+Sphinx
+======
+
+.. cssclass:: sphinx-tagline
```

Comment:
I think the old text was too compact but now the boxes feel too large. I'd be happy if you added more links to the "versatile format" section, where you would link to the builders as well.

For the automatic docs section, I'd put "Python" and "C++" links to their corresponding domain. For the extensions section I'd link the "creating diagrams" to the corresponding extension as well.

Overall, I'd prefer having quite a lot of links in the boxes at the top so that I don't need to go through the TOC or something else. I think it would be more accessible in general (though I'd like to avoid having too many links otherwise the text becomes unreadable).

I think it'd be easier for me to review the pages if you split the PRs though.

Reply:
Actually the pages were small enough so no need to split into multiple PRs.

Reply:
> I think the old text was too compact but now the boxes feel too large, ...

I changed the text for each feature to be more imperative and concise, and added links to the main `Builders` and `Domains` page in the relevant ones.

> I'd link the "creating diagrams" to the corresponding extension as well

I think the link to the main `Extensions` page is sufficient here, as you say we don't want too many links 😅